### PR TITLE
support VCA cached for CACHE_CAP in 1.2 for transcript.

### PR DESCRIPTION
**Pull request by qizhangz**
Fix: #784 

Signed-off-by: Qi Zhang <qi1.zhang@intel.com>

### CODE REVIEW

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 1]
```diff
@@ -758,6 +766,10 @@ libspdm_return_t libspdm_get_data(void *context, libspdm_data_type_t data_type,
         target_data_size = sizeof(bool);
         target_data = &spdm_context->need_reset_to_get_csr;
         break;
+    case LIBSPDM_DATA_VCA_CACHE:
+        target_data_size = sizeof(libspdm_small_managed_buffer_t);
```
Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:770]: target_data_size shou;d be the buffer_size.

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 2]
```diff
@@ -511,6 +511,14 @@ libspdm_return_t libspdm_set_data(void *context, libspdm_data_type_t data_type,
         }
         spdm_context->need_reset_to_get_csr = *(bool *)data;
         break;
+    case LIBSPDM_DATA_VCA_CACHE:
+        if (data_size != sizeof(libspdm_small_managed_buffer_t)) {
+            return LIBSPDM_STATUS_INVALID_PARAMETER;
+        }
+        libspdm_copy_mem(&spdm_context->transcript.message_a,
+                         sizeof(spdm_context->transcript.message_a),
+                         data, data_size);
```
Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:520]: Please only copy the buffer out, not managed buffer

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 3]
```diff
@@ -758,6 +766,10 @@ libspdm_return_t libspdm_get_data(void *context, libspdm_data_type_t data_type,
         target_data_size = sizeof(bool);
         target_data = &spdm_context->need_reset_to_get_csr;
         break;
+    case LIBSPDM_DATA_VCA_CACHE:
+        target_data_size = sizeof(spdm_context->transcript.message_a.buffer);
```
Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:770]: This is wrong, please just return actual size. not max_size.

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 4]
```diff
@@ -511,6 +511,14 @@ libspdm_return_t libspdm_set_data(void *context, libspdm_data_type_t data_type,
         }
         spdm_context->need_reset_to_get_csr = *(bool *)data;
         break;
+    case LIBSPDM_DATA_VCA_CACHE:
+        if (data_size != sizeof(spdm_context->transcript.message_a.buffer)) {
```
Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:515]: This is wrong, please just return actual size. not max_size.

**Review by jyao1 — APPROVED** [review]

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**